Update interprocess.md (#19670)
dotnet/AspNetCore.Docs: aspnetcore/grpc/interprocess.md

By [James Newton-King](https://twitter.com/jamesnk)

gRPC calls between a client and service are usually sent over TCP sockets. TCP is great for communicating across a network, but [inter-process communication (IPC)](https://wikipedia.org/wiki/Inter-process_communication) is more efficient when the client and service are on the same machine. This document explains how to use gRPC with custom transports in IPC scenarios.

## Server configuration

> Custom transports are supported by Kestrel. Endpoints are configured in *Program.cs*:
> 
> * Kestrel endpoints with `ConfigureKestrel`.
> * Custom endpoints, such as [Unix domain sockets](https://wikipedia.org/wiki/Unix_domain_socket).
> 
> Kestrel supports listening on unix sockets with <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenUnixSocket*>.

Reviewer: ```suggestion
Custom transports are supported by Kestrel. Kestrel is configured in *Program.cs*:
```

        });
```

> Unix domain sockets are [supported](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/) by modern versions of Windows.

Reviewer: ```suggestion
The preceding example:

* Configures Kestrel endpoints in `ConfigureKestrel`.
* Calls <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenUnixSocket*> to listen to a [Unix domain socket (UDS)](https://en.wikipedia.org/wiki/Unix_domain_socket) with the specified path.

Kestrel has built-in support for UDS endpoints. UDS are supported on Linux, macOS and [modern versions of Windows](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/).
```
